examples/misc/anchored_artists.py

```diff
@@ -9,76 +9,42 @@
 of additional toolkits.
+
+.. redirect-from:: /gallery/userdemo/anchored_box01
+.. redirect-from:: /gallery/userdemo/anchored_box02
+.. redirect-from:: /gallery/userdemo/anchored_box03
 """
 
 from matplotlib import pyplot as plt
 from matplotlib.lines import Line2D
-from matplotlib.patches import Ellipse
+from matplotlib.patches import Circle, Ellipse
 from matplotlib.offsetbox import (
     AnchoredOffsetbox, AuxTransformBox, DrawingArea, TextArea, VPacker)
 
 
-class AnchoredText(AnchoredOffsetbox):
-    def __init__(self, s, loc, pad=0.4, borderpad=0.5,
-                 prop=None, frameon=True):
-        self.txt = TextArea(s)
-        super().__init__(loc, pad=pad, borderpad=borderpad,
-                         child=self.txt, prop=prop, frameon=frameon)
-
-
 def draw_text(ax):
-    """
-    Draw a text-box anchored to the upper-left corner of the figure.
-    """
-    at = AnchoredText("Figure 1a", loc='upper left', frameon=True)
-    at.patch.set_boxstyle("round,pad=0.,rounding_size=0.2")
-    ax.add_artist(at)
-
-
-class AnchoredDrawingArea(AnchoredOffsetbox):
-    def __init__(self, width, height, xdescent, ydescent,
-                 loc, pad=0.4, borderpad=0.5, prop=None, frameon=True):
-        self.da = DrawingArea(width, height, xdescent, ydescent)
-        super().__init__(loc, pad=pad, borderpad=borderpad,
-                         child=self.da, prop=None, frameon=frameon)
-
-
-def draw_circle(ax):
-    """
-    Draw a circle in axis coordinates
-    """
-    from matplotlib.patches import Circle
-    ada = AnchoredDrawingArea(20, 20, 0, 0,
-                              loc='upper right', pad=0., frameon=False)
-    p = Circle((10, 10), 10)
-    ada.da.add_artist(p)
-    ax.add_artist(ada)
+    """Draw a text-box anchored to the upper-left corner of the figure."""
+    box = AnchoredOffsetbox(child=TextArea("Figure 1a"),
+                            loc="upper left", frameon=True)
+    box.patch.set_boxstyle("round,pad=0.,rounding_size=0.2")
+    ax.add_artist(box)
 
 
-class AnchoredEllipse(AnchoredOffsetbox):
-    def __init__(self, transform, width, height, angle, loc,
-                 pad=0.1, borderpad=0.1, prop=None, frameon=True):
-        """
-        Draw an ellipse the size in data coordinate of the give axes.
-
-        pad, borderpad in fraction of the legend font size (or prop)
-        """
-        self._box = AuxTransformBox(transform)
-        self.ellipse = Ellipse((0, 0), width, height, angle)
-        self._box.add_artist(self.ellipse)
-        super().__init__(loc, pad=pad, borderpad=borderpad,
-                         child=self._box, prop=prop, frameon=frameon)
+def draw_circles(ax):
+    """Draw circles in axes coordinates."""
+    box = AnchoredOffsetbox(child=DrawingArea(40, 20, 0, 0),
+                            loc="upper right", pad=0., frameon=False)
+    box.get_child().add_artist(Circle((10, 10), 10, fc="tab:blue"))
+    box.get_child().add_artist(Circle((30, 10), 5, fc="tab:red"))
```

Comment:
```suggestion
    drawing_area = DrawingArea(40, 20, 0, 0)
    drawing_area.add_artist(Circle((10, 10), 10, fc="tab:blue"))
    drawing_area.add_artist(Circle((30, 10), 5, fc="tab:red"))
    box = AnchoredOffsetbox(child=drawing_area,
                            loc="upper right", pad=0., frameon=False)
```
It's clearer to keep a reference to the `DrawingArea` explicitly instead of getting it back via `box.get_child()`.

Reply:
sure